src/query/models/query_context.go

```diff
@@ -48,6 +48,8 @@ type QueryContextOptions struct {
 	LimitMaxDocs int
 	// RequireExhaustive results in an error if the query exceeds the series limit.
 	RequireExhaustive bool
+	// Is instant query
+	Instantaneous     bool
 	RestrictFetchType *RestrictFetchTypeQueryContextOptions
```

Comment:
nit: make these godoc style comments;
```suggestion
	// Instantaneous indicates an instant query.
	Instantaneous     bool
	// RestrictFetchType restricts the query fetches.
	RestrictFetchType *RestrictFetchTypeQueryContextOptions
```